src/core/providers/ogr/qgsogrprovider.cpp

```diff
@@ -1222,72 +1222,42 @@ QgsBox3D QgsOgrProvider::extent3D() const
           mExtent->MaxZ = std::numeric_limits<double>::quiet_NaN();
           hasBeenComputed = true;
         }
+        // else slow computation
       }
-      else
+      else if ( mSubsetString.isEmpty() ) // TODO: find a way to optimize when mSubsetString is defined
       {
-        if ( is3D && mSubsetString.isEmpty() ) // TODO: find a way to use mSubsetString in the request
+        QgsDebugMsgLevel( QStringLiteral( "WITH OLCFastGetExtent" ), 3 );
+        QRecursiveMutex *mutex = nullptr;
+        OGRLayerH layer = mOgrLayer->getHandleAndMutex( mutex );
+        const char *geomCol = OGR_L_GetGeometryColumn( layer );
+        if ( geomCol != NULL && strlen( geomCol ) > 0 )
```

Comment:
you don't need that test anymore as you don't write SQL. You could just test OGR_L_GetGeomType() != wkbNone

Reply:
ok fixed